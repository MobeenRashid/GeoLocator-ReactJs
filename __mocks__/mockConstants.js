import { arrayToMap } from "../src/utils/arrayUtils";

const __mockLocationList = [
    {
        id: '1',
        address: 'Berlin',
        lng: "123",
        lat: "456"
    },
    {
        id: '2',
        address: 'Munich',
        lng: "124",
        lat: "457"
    }
];


const __mockState = {
    locations: __mockLocationList,
    requestingLocations: false
}

export { __mockState, __mockLocationList };