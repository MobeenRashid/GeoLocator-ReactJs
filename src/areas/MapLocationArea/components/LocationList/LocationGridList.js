import React from 'react';
import { List, Card, Icon, Popconfirm, Spin } from 'antd';
import { Link } from 'react-router-dom';


const LocationGridList = (props) => (
    <Spin tip="loading..." spinning={props.fetchingData}>
        <div>
            <List className="location-list" grid={{ gutter: 10, md: 3, sm: 2, xs: 2 }}
                dataSource={props.data}
                renderItem={item => (
                    <List.Item className="location-item">
                        <Card title={item.address} className="location-card"
                            actions={
                                [
                                    <Link to={`locations/edit/${item.id}`}>
                                        <Icon type="edit" />
                                    </Link>,
                                    <Popconfirm title="Are you sure you want to delete the location?" okText="yes" cancelText="no" onConfirm={() => { props.onLocationDelete(item.id) }}>
                                        <Icon type="delete" />
                                    </Popconfirm>]
                            }>
                            <p className="nowrap-ellip"><b>{item.address}</b></p>
                            <p className="nowrap-ellip">Latitude: {item.lat}</p>
                            <p className="nowrap-ellip">Longitude: {item.lng}</p>
                        </Card>
                    </List.Item>
                )}
            />
        </div>
    </Spin>
);


export default LocationGridList;