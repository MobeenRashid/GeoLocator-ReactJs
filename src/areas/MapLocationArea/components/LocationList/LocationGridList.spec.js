import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom'
import LocationGridList from './LocationGridList'
import { __mockLocationList } from '../../../../../__mocks__/__mockState';


describe("LocationGridList component", () => {
    test('Should render correctly', () => {
        const component = mount(<LocationGridList />);
        expect(component).toMatchSnapshot();
        component.unmount();
    });

    test('Should render location list corretly', () => {
        const component = mount(
            <BrowserRouter>
                <LocationGridList data={__mockLocationList} />
            </BrowserRouter>
        );

        const gridListComponent = component.find(LocationGridList);
        expect(gridListComponent.props().data).toBe(__mockLocationList);

        //Heading:asserts to verfiy address
        const headTitles = component.find('.ant-card-head-title');

        expect(headTitles.get(0).props.children).toBe('Berlin');
        expect(headTitles.get(1).props.children).toBe('Munich');

        const bodyContents = component.find('.ant-card-body');
        const card1Body = bodyContents.get(0);
        const card2Body = bodyContents.get(1);

        //Body:asserts to verify address
        const address1 = card1Body.props.children[0];
        const address2 = card2Body.props.children[0];
        expect(address1.props.children.props.children).toBe('Berlin');
        expect(address2.props.children.props.children).toBe('Munich');

        //Body:asserts to verify latitude
        const lat1 = card1Body.props.children[1];
        const lat2 = card2Body.props.children[1];

        expect(lat1.props.children[1]).toBe("456");
        expect(lat2.props.children[1]).toBe("457");


        // //Body:asserts to verify longitude 
        const lng1 = card1Body.props.children[2];
        const lng2 = card2Body.props.children[2];

        expect(lng1.props.children[1]).toBe("123");
        expect(lng2.props.children[1]).toBe("124");

        component.unmount();
    });
});