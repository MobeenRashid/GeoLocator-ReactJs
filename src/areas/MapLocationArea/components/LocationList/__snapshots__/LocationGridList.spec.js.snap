// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationGridList component Should render correctly 1`] = `
<LocationGridList>
  <Spin
    size="default"
    spinning={true}
    tip="loading..."
    wrapperClassName=""
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        key="loading"
      >
        <div
          className="ant-spin ant-spin-spinning ant-spin-show-text"
        >
          <span
            className="ant-spin-dot ant-spin-dot-spin"
          >
            <i />
            <i />
            <i />
            <i />
          </span>
          <div
            className="ant-spin-text"
          >
            loading...
          </div>
        </div>
      </div>
      <div
        className="ant-spin-container ant-spin-blur"
        key="container"
      >
        <div>
          <List
            bordered={false}
            className="location-list"
            dataSource={Array []}
            grid={
              Object {
                "gutter": 10,
                "md": 3,
                "sm": 2,
                "xs": 2,
              }
            }
            loading={false}
            pagination={false}
            renderItem={[Function]}
            split={true}
          >
            <div
              className="ant-list location-list ant-list-split ant-list-grid"
            >
              <Spin
                size="default"
                spinning={false}
                wrapperClassName=""
              >
                <div
                  className="ant-spin-nested-loading"
                >
                  <div
                    className="ant-spin-container"
                    key="container"
                  >
                    <div
                      className="ant-list-empty-text"
                    >
                      <Empty
                        className="ant-empty-normal"
                        image="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                      >
                        <LocaleReceiver
                          componentName="Empty"
                        >
                          <div
                            className="ant-empty ant-empty-normal"
                          >
                            <div
                              className="ant-empty-image"
                            >
                              <img
                                alt="No Data"
                                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNDEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMCAxKSIgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJldmVub2RkIj4KICAgIDxlbGxpcHNlIGZpbGw9IiNGNUY1RjUiIGN4PSIzMiIgY3k9IjMzIiByeD0iMzIiIHJ5PSI3Ii8+CiAgICA8ZyBmaWxsLXJ1bGU9Im5vbnplcm8iIHN0cm9rZT0iI0Q5RDlEOSI+CiAgICAgIDxwYXRoIGQ9Ik01NSAxMi43Nkw0NC44NTQgMS4yNThDNDQuMzY3LjQ3NCA0My42NTYgMCA0Mi45MDcgMEgyMS4wOTNjLS43NDkgMC0xLjQ2LjQ3NC0xLjk0NyAxLjI1N0w5IDEyLjc2MVYyMmg0NnYtOS4yNHoiLz4KICAgICAgPHBhdGggZD0iTTQxLjYxMyAxNS45MzFjMC0xLjYwNS45OTQtMi45MyAyLjIyNy0yLjkzMUg1NXYxOC4xMzdDNTUgMzMuMjYgNTMuNjggMzUgNTIuMDUgMzVoLTQwLjFDMTAuMzIgMzUgOSAzMy4yNTkgOSAzMS4xMzdWMTNoMTEuMTZjMS4yMzMgMCAyLjIyNyAxLjMyMyAyLjIyNyAyLjkyOHYuMDIyYzAgMS42MDUgMS4wMDUgMi45MDEgMi4yMzcgMi45MDFoMTQuNzUyYzEuMjMyIDAgMi4yMzctMS4zMDggMi4yMzctMi45MTN2LS4wMDd6IiBmaWxsPSIjRkFGQUZBIi8+CiAgICA8L2c+CiAgPC9nPgo8L3N2Zz4K"
                              />
                            </div>
                            <p
                              className="ant-empty-description"
                            >
                              No Data
                            </p>
                          </div>
                        </LocaleReceiver>
                      </Empty>
                    </div>
                  </div>
                </div>
              </Spin>
            </div>
          </List>
        </div>
      </div>
    </div>
  </Spin>
</LocationGridList>
`;
